### ISSUE TITLE
Missing nullability warnings for semi-autoproperties

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#76487]
**Version Used**: 
```
Branch main (17 Dec 2024)
Latest commit [f9ccce0](https://github.com/dotnet/roslyn/commit/f9ccce030d8fdcc2ea095e095b71316d243c5b35) by Phil Allen:
Update PublishData.json (#76471)
```

**Steps to Reproduce**:

Compile and run the following code ([sharplab.io](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUPgAwAE+xAdACJQQDmMA9gM4JQDG3BgGFOGAKYBBGBAA2AT25RuAbgoBiGKlmyIwWeJriZ+8RQoA3CMhoAHGgF4aMcQHcawgBQBKBgAVkTls1SmIATk9bBgAZYzYEAAtvEIoiDwoAbwoaGgBtADlOBHztWQBdbLoAZjpiKgB+GgCgyqzyHJy2cQRHAD4aADMocVkMEI6abm6+weHRxxorWVRDesb6KgYAUTBbBHlxmgBfBa0dEKOgA===))
```cs
using System;
using System.Diagnostics.CodeAnalysis;
#nullable enable

var p = new C().Prop;
Console.WriteLine(p.Length);

class C
{
  [NotNull]
  public string? Prop
  {
    get;
    set => field = value ?? string.Empty;
  } = null;
}
```

**Expected Behavior**:
A warning is emitted for `get;` accessor that returns a possibly nullable field value as a result of a non-nullable property

**Actual Behavior**:
No warnings at all in the program above. It crashes with a `NullReferenceException` at runtime

**Notes**:
If you explicitly implement `get;` with the equivalent `get => field;` you get the correct `CS8603: Possible null reference return.` warning

**Comment by cston** [comment]
For comparison, no warnings are reported for full auto-properties (`P1`, `P2` below), although warnings are reported for the equivalent explicit `field` implementations (`P3`, `P4`) as mentioned above - see [sharplab.io](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEAMACdBGAOgBEoIBzGAewGcAXKAY2oIGFKEBTAQRggBsAntSjUA3AFgAUAGIYAVz58IwPh2wdeKjlKnoATNhZSA3lOznsAbQBylWtYV8AumYvoAzLjyYA/NgAKeK7mppIW4dhkHLSi2AD0cdhU2ADuEABOMLBkwdgAvtgAvEmOEmEWuTZ2Doou5eYeXr4BermhERZRMfGJyWmZ2bnh1NFl4QXF8opjFfVWtvaOdeGN+M3+7m1DndFFAHzYAGZQHHwIsQmpGVkwZIYAygAcAGyY7iABNMJaJYrY6RxDhwATAGGp0AB2XITX58GbmSoLGrOXKrbx+fwAFi2c3CXX2RxOZwuiX6NzuLCer3en2o31UsP+gOBGjBuChuIsI1o2GMhNOCCK2AAbvw5BxYnloUKpnCpHkgA=).

```csharp
using System.Diagnostics.CodeAnalysis;
#nullable enable

class C
{
    [NotNull]
    public string? P1
    {
        get; // no warning
    } = null;
    
    [NotNull]
    public string? P2
    {
        get; // no warning
        set;
    } = null;
    
    [NotNull]
    public string? P3
    {
        get => field; // warning CS8603: Possible null reference return
    } = null;
    
    [NotNull]
    public string? P4
    {
        get => field; // warning CS8603: Possible null reference return
        set { field = value; }
    } = null;
}
```

**Comment by RikkiGibson** [comment]
Yeah, the code elements involved in the missing warning are just the property initializer and get accessor, so, this is an old bug. I opted not to fix it when doing the initial field nullability pass, but we could take the opportunity this time if we want.
